packages/dara-components/changelog.md

```diff
@@ -2,6 +2,23 @@
 title: Changelog
 ---
 
+## NEXT
+
+-   Updated AnnotatedAction to have a `loading` property that is automatically set to be a Variable[bool] instance tracking the loading state of the action. The example below shows how you can disable a button whilst the action it triggers is running.
+
+```python
+from dara.core import action
+from dara.components import Button
+
+@action
+def on_click(ctx: action.Ctx):
+    # Do Something...
+
+Button('Click Me', onclick=on_click, disabled=on_click.loading)
```

Comment:
Shouldn't it be this way:

```suggestion
@action
def on_click(ctx: action.Ctx):
    # Do Something...
    
action_instance = on_click()

Button('Click Me', onclick=action_instance, disabled=action_instance.loading)
```

Since you can invoke the same action creator multiple times, they get separate loading states right?

Reply:
Ah, yeah fair point. I wasn't sure at what point it became an annotated action. Will update this.